lib/aerosol/auto_scaling.rb

```diff
@@ -38,21 +39,30 @@ def exists?
   end
 
   def create!
-    ensure_present! :launch_configuration, :max_size, :min_size
+    ensure_present! :max_size, :min_size
     raise 'availability_zones or vpc_zone_identifier must be set' if [availability_zones, vpc_zone_identifier].none?
+    raise 'launch_configuration or launch_template must be set' if [launch_configuration, launch_template].none?
 
     info "creating auto scaling group"
-    launch_configuration.create
+    launch_details = create_launch_details
+
     info self.inspect
 
     conn.create_auto_scaling_group({
       auto_scaling_group_name: auto_scaling_group_name,
       availability_zones: [*availability_zones],
-      launch_configuration_name: launch_configuration.launch_configuration_name,
       max_size: max_size,
       min_size: min_size
-    }.merge(create_options))
-    sleep 10
+    }
+      .merge(create_options)
+      .merge(launch_details)
+    )
+
+    conn.wait_until(:group_in_service, auto_scaling_group_names: [auto_scaling_group_name]) do |waiter|
+      waiter.before_wait do |attempt_count, _response|
+        info "Wait count #{attempt_count} of #{waiter.max_attempts} for #{auto_scaling_group_name} to be in service"
```

Comment:
What does `in service` mean here? Just like, persisted essentially? I see we're replacing the `sleep 10`.

Reply:
Ah! Yes, 
```
On the Instance management tab, under Instances, you can view the status of the instances that are currently running. The Lifecycle column contains the state of your instances. Note that it takes a short time for an instance to launch. After the instance starts, its lifecycle state changes to InService.
```
All instances have started. The sleep 10 is fine if that's what we want. The InService check takes about 15-20 seconds.

Reply:
![image](https://user-images.githubusercontent.com/29490857/107514380-9630e880-6b77-11eb-8c78-296000a5d852.png)